demos/guide/README.md

```diff
@@ -5,11 +5,73 @@ The demos in this directory are the complete source code for the applications
 developed in the OpenSSL Guide tutorials. Refer to the various tutorial pages in
 the [guide] for an extensive discussion on the demos available here.
 
+They must be built before they can be run. An example UNIX style Makefile is
+supplied. Just type "make" from this directory on a Linux/UNIX system.
+
+Running the TLS Demos
+---------------------
+
 To run the demos when linked with a shared library (default) ensure that
 libcrypto and libssl are on the library path. For example, assuming you have
 already built OpenSSL from this source and in the default location then to run
 the tls-client-block demo do this:
 
-LD_LIBRARY_PATH=../.. ./tls-client-block
+LD_LIBRARY_PATH=../.. ./tls-client-block hostname port
+
+In the above replace "hostname" and "port" with the hostname and the port number
+of the server you are connecting to.
+
+The above assumes that your default trusted certificate store containing trusted
+CA certificates has been properly setup and configured as described on the
+[TLS Introduction] page.
+
+You can run a test server to try out these demos using the "openssl s_server"
+command line utility and using the test server certificate and key provided in
+this directory. For example:
+
+openssl s_server -www -accept localhost:4443 -cert servercert.pem -key serverkey.pem
```

Comment:
Should this be wrapped?  `openssl` will pick up the system version if present and nothing if not.

Reply:
I added `LD_LIBRARY_PATH` etc to make this work